src/main/java/tools/jackson/databind/deser/bean/BeanDeserializer.java

```diff
@@ -915,6 +919,15 @@ protected Object deserializeWithUnwrapped(JsonParser p, DeserializationContext c
                 handleIgnoredProperty(p, ctxt, bean, propName);
                 continue;
             }
+
+            // [databind#1709]: Track non-null unwrapped property
+            if (unwrappedPropertyNames != null
```

Comment:
Remove null checks: `unwrappedPropertyNames` must be non null (we wouldn't get here otherwise); and `propName` likewise must be non-null.